collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_network.py

```diff
@@ -0,0 +1,240 @@
+            if module.params['gateway'] not in IPNetwork(full_addr):
+                module.fail_json(msg='Gateway and subnet are not compatible.')
+            gateway = module.params['gateway']
+        new_state = {'address': address,
+                     'mtu': mtu,
+                     'gateway': gateway,
+                     'netmask': netmask}
+        if new_state == current_state:
+            changed = False
+        else:
+            if 'management' in interface['services'] or 'app' in interface['services'] and address == "0.0.0.0/0":
+                module.fail_json(msg="Removing IP address from a management or app port is not supported")
+            try:
+                array.set_network_interface(interface['name'],
+                                            address=new_state['address'],
+                                            mtu=new_state['mtu'],
+                                            netmask=new_state['netmask'],
+                                            gateway=new_state['gateway'])
+            except Exception:
+                module.fail_json(msg="Failed to change settings for interface {0}.".format(interface['name']))
+        if not interface['enabled'] and module.params['state'] == 'present':
+            try:
+                array.enable_network_interface(interface['name'])
+                changed = True
+            except Exception:
+                module.fail_json(msg="Failed to enable interface {0}.".format(interface['name']))
+
+    module.exit_json(changed=changed)
+
+
+def disable_interface(module, array):
+    """ Disable Interface"""
+    changed = True
+    if not module.check_mode:
+        if module.params['name'][0] == "v":
+            try:
+                array.disable_network_interface(module.params['name'])
+            except Exception:
+                module.fail_json(msg="Failed to disable inerface {0}".format(module.params['name']))
+        else:
+            try:
+                array.disable_network_interface(module.params['name'])
+            except Exception:
+                module.fail_json(msg="Failed to disable inerface {0}.".format(module.params['name']))
+    module.exit_json(changed=changed)
+
+
+def main():
+    argument_spec = purefa_argument_spec()
+    argument_spec.update(
+        dict(
+            name=dict(type=str, required=True),
+            state=dict(type=str, default='present', choices=['present', 'absent']),
+            address=dict(type=str),
+            gateway=dict(type=str),
+            mtu=dict(type='int', default=1500),
+        )
+    )
+
+    module = AnsibleModule(argument_spec,
+                           supports_check_mode=True)
+
+    if not HAS_NETADDR:
+        module.fail_json(msg='netaddr module is required')
+
+    state = module.params['state']
+    if '.' in module.params['name'] or '.' in module.params['controller']:
+        module.fail_json(msg='Incorrect controller or interface provided.')
```

Comment:
Lines 223 and 224 should be removed, since controller is no longer a valid param and name can include a '.' now.